tests/Tests.AzureAppConfiguration/RefreshTests.cs

```diff
@@ -101,10 +121,41 @@ public void RefreshTests_RefreshIsSkippedIfCacheIsNotExpired()
                 .AddAzureAppConfiguration(options =>
                 {
                     options.Client = mockClient.Object;
-                    options.Select("TestKey*");
                     options.ConfigureRefresh(refreshOptions =>
                     {
-                        refreshOptions.Register("TestKey1")
+                        refreshOptions.Register("TestKey1", "label")
+                            .SetCacheExpiration(TimeSpan.FromSeconds(10));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            Assert.Equal("TestValue1", config["TestKey1"]);
+            FirstKeyValue.Value = "newValue1";
+
+            // Wait for some time but not enough to let the cache expire
+            Thread.Sleep(5000);
```

Comment:
It won't let me put a comment in `RefreshTests_RefreshIsSkippedIfKvNotInSelectAndCacheIsNotExpired`, but we don't need a sleep there right? The issue it is trying to test is that we don't skip keys not in select.

Reply:
RefreshTests_RefreshIsSkippedIfKvNotInSelectAndCacheIsNotExpired is testing the same functionality as RefreshTests_RefreshIsSkippedIfCacheIsNotExpired. 
The only difference is that in RefreshIsSkippedIfCacheIsNotExpired, the refresh registered key is already loaded (default behavior is to load all keys when `Select` clause is missing), whereas in RefreshIsSkippedIfKvNotInSelectAndCacheIsNotExpired, the refresh registered key is not already loaded by the `Select` clause.

---
In reply to: [500698506](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/204#discussion_r500698506) [](ancestors = 500698506)

Reply:
Any  test that uses arbitrary numbers for delays is suspect to me. I would say that both of these tests would be fine without the sleep at all.

Reply:
Makes sense, I'll remove sleep from both tests. Thanks!

---
In reply to: [501235350](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/204#discussion_r501235350) [](ancestors = 501235350)